### DISCUSSION TITLE
Unable to call SQL function from Edge Function (returns error)

### DISCUSSION THREAD

**Discussion by vv001** [supabase/supabase#29651] [Questions] [answered]
I have the following SQL Function:

```
CREATE OR REPLACE FUNCTION public.check_user_exists(email_to_check TEXT)
    RETURNS UUID
    LANGUAGE plpgsql
    SECURITY DEFINER
    SET search_path = public
AS $$
DECLARE
    user_id UUID;
BEGIN
    SELECT id INTO user_id
    FROM auth.users
    WHERE email = email_to_check
    LIMIT 1;

    RETURN user_id;
END;
$$;

```

When I call the function from Supabase JS in my Angular application through the following code, it works perfectly:
```
    public async checkUserAvailability(email: string) {
        const { data, error } = await this.supabaseService.supabaseClient
            .rpc('check_user_exists', {
                email_to_check: email
            })

        if (error) {
            throw new Error("Failed to check user availability, because: " + error.message);
        }

        return data;
    }
```

I have a Supabase Edge function (deployed), from which I call the above SQL Function, but it shows an error, which is confusing to me:
```
async function checkUserExists(params: { email: string, token: string, role: string }) {
  console.log('Checking user exists for email:', params.email);

  try {
    console.log('Supabase client config:', supabase.config);

    const { data, error } = await supabase
        .rpc('check_user_exists', { email_to_check: params.email });

    console.log('RPC call completed');

    if (error) {
      console.error('Error in checkUserExists:', JSON.stringify(error, null, 2));
      throw error;
    }

    console.log('checkUserExists result:', data);
    return data;
  } catch (error) {
    console.error('Caught error in checkUserExists:', JSON.stringify(error, null, 2));
    throw error;
  }
}
```

This is the error:
**Error processing request: {
  code: "PGRST202",
  details: "Searched for the function public.check_user_exists without parameters or with a single unnamed json/jsonb parameter, but no matches were found in the schema cache.",
  hint: null,
  message: "Could not find the function public.check_user_exists without parameters in the schema cache"
}**

How can it be that calling it with Supabase JS from Angular web application works, but calling it from a deployed Edge Function returns an error?

**Comment by GaryAustin1** [comment, marked as answer]
It is acting like it is calling the function without parameters from your call.  Maybe try and hard code an email string to test if your parameter is somehow the issue.  I would expect a different error if it could not find the function at all.

**Comment by vv001** [comment]
It seems that something was going wrong with the parameter, which was 'undefined'. Now it's fixed and it is all working again. Thanks for the help @GaryAustin1
